### PR TITLE
Openmdao fix

### PR DESCRIPTION
**Pull request by christophe-david**
This PR gets back compatibility with current OpenMDAO dev version after PR https://github.com/OpenMDAO/OpenMDAO/pull/2655/

Independent vars are now identified by having either the old tag `"indep_var"` or the new one `"openmdao:indep_var"`.

Watchman tests will still fail because of the missing `"grid"` format in `OptionsDictionary.to_table()`, but that is a work in progress in OpenMDAO team.

### CODE REVIEW

**Review by ScottDelbecq — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -46,7 +46,7 @@ python = "^3.7"
 numpy = "^1.21.0"
 scipy = "^1.4.1"
 pandas = "^1.1.0"
-openmdao = ">=3.10, <=3.21"
+openmdao = ">^3.10"
```
Comment (ScottDelbecq) [inline pyproject.toml:49]:
```suggestion
openmdao = "^3.10"
```

**Review by ScottDelbecq — CHANGES_REQUESTED** [review]
There is a small typo in the pyproject.toml which leads to failing to build the .lock in Watchmen tests.

File: pyproject.toml (hunk 1 above)
Comment (christophe-david) [reply pyproject.toml]: Thanks. Fixed in afbbd1eb.
The watchman tests now fail at unit tests, as expected 😄

Comment (christophe-david) [reply pyproject.toml]: For the record, the typo didn't bothered `poetry lock --no-update` 😅

Comment (ScottDelbecq) [reply pyproject.toml]: Indeed 😄

**Review by ScottDelbecq — APPROVED** [review]

**Pull request merged**